src/htmlize/patchtml.ml

```diff
@@ -20,10 +19,115 @@ let edit_html ?header ?footer ~head_childs html =
     footer
     "#footer";
   let head = html $ "head" in
-  List.iter (fun (id,child) -> 
-      opt_inserter
-        (iter (fun e -> append_child head e))
-        (Some child)
-        id)
+  List.iter (fun (id,child) ->
+    opt_inserter
+      (iter (fun e -> append_child head e))
+      (Some child)
+      id)
     head_childs;
   to_string html
+
+open Omd
+(* Keeps us from processing url links *)
+let is_url url =
+  (String.length url >= 7 && String.equal (String.sub url 0 7) "http://")
+  || (String.length url >= 8 && String.equal (String.sub url 0 8) "https://")
+
+(* Keeps us from processing anchors *)
+let is_anchor link =
+  String.length link >= 1 && String.equal (String.sub link 0 1) "#"
+
+(* Establish depth of given link relative to the root
+ * Then prepend the corresponding number of .. to the basename
+ * Then append index.html to fix link
+ * *)
+let handle_link link =
+  if link.destination = "" then
+    link
+
+  (* Escape through this branch if the link points to an http(s) address,
+   * or is a link from a Table of Contents
+   * *)
+  else if is_url link.destination || is_anchor link.destination then
```

Comment:
I believe you can merge this with the previous case :-)

Reply:
Actually, I tried on gitea and in case of an empty link, they turn it into text and remove the link. On GitHub it fails (the same way it fails for `.` or `./`) so maybe we don't want to return a link in this case but turn it to text too ?

Reply:
I've programmed it to return a text only if the link is `""` and the constructor is `Link`

Reply:
OK so now it shouldn't happen ? You could replace it by an `assert false` or simply remove the first `if`.

Reply:
Since i've removed the necessity of first if with the adding of `construct` function, I've removed it from the `handle_link` function and let the empty list match case as is